spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/serializer/TcpSerDe.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.ip.tcp.serializer;
+
+/**
+ * Factory class to create TCP Serializer/Deserializers.
+ * Used to simplify configuration with Java, such as
+ *
+ * <pre class="code">
+ * TcpNetServerConnectionFactory server = new TcpNetServerConnectionFactory(1234);
+ * server.setSerializer(TcpSerDe.lf());
+ * server.setDserializer(TcpSerDe.lf());
+ * ...
+ * </pre>
+ *
+ * @author Gary Russell
+ * @since 5.0
+ *
+ */
+public final class TcpSerDe {
```

Comment:
I don't like this name for top level API.
Why `TcpSerializers` doesn't work for you?

Looks like JavaDoc covers the purpose if that.

OTOH, any target implementation doesn't mention in their name `de` functionality, and just called like `AbstractByteArraySerializer`.

Am I missing something?

Otherwise this PR has good premise and will be merged when we reach the compromise about names :smile: